src/test/integration/org/apache/cassandra/sidecar/routes/BaseTokenRangeIntegrationTest.java

```diff
@@ -0,0 +1,257 @@
+                                                                   dcAndRack("datacenter2", "rack2")));
+        });
+    }
+
+    protected List<Range<BigInteger>> generateExpectedRanges()
+    {
+        CassandraIntegrationTest annotation = sidecarTestContext.cassandraTestContext().annotation;
+        int nodeCount = (annotation.nodesPerDc() + annotation.newNodesPerDc()) * annotation.numDcs();
+        return generateExpectedRanges(nodeCount);
+    }
+
+    protected List<Range<BigInteger>> generateExpectedRanges(int nodeCount)
+    {
+        CassandraIntegrationTest annotation = sidecarTestContext.cassandraTestContext().annotation;
+        TokenSupplier tokenSupplier = (annotation.numDcs() > 1) ?
+                                      MultiDcTokenSupplier.evenlyDistributedTokens(
+                                      annotation.nodesPerDc() + annotation.newNodesPerDc(),
+                                      annotation.numDcs(),
+                                      1) :
+                                      TokenSupplier.evenlyDistributedTokens(annotation.nodesPerDc() +
+                                                                            annotation.newNodesPerDc(),
+                                                                            1);
+
+        List<Range<BigInteger>> expectedRanges = new ArrayList<>();
+        BigInteger startToken = Partitioner.Murmur3.minToken;
+        BigInteger endToken = Partitioner.Murmur3.maxToken;
+        int node = 1;
+        BigInteger prevToken = new BigInteger(tokenSupplier.tokens(node++).stream().findFirst().get());
+        Range<BigInteger> firstRange = Range.openClosed(startToken, prevToken);
+        expectedRanges.add(firstRange);
+        while (node <= nodeCount)
+        {
+            BigInteger currentToken = new BigInteger(tokenSupplier.tokens(node).stream().findFirst().get());
+            expectedRanges.add(Range.openClosed(prevToken, currentToken));
+            prevToken = currentToken;
+            node++;
+        }
+        expectedRanges.add(Range.openClosed(prevToken, endToken));
+        return expectedRanges;
+    }
+
+    protected Set<String> instancesFromReplicaSet(List<TokenRangeReplicasResponse.ReplicaInfo> replicas)
+    {
+        return replicas.stream()
+                       .flatMap(r -> r.replicasByDatacenter().values().stream())
+                       .flatMap(Collection::stream)
+                       .collect(Collectors.toSet());
+    }
+
+    protected void validateWriteReplicaMappings(List<TokenRangeReplicasResponse.ReplicaInfo> writeReplicas,
+                                              Map<String, Map<Range<BigInteger>, List<String>>> expectedRangeMapping)
+    {
+        CassandraIntegrationTest annotation = sidecarTestContext.cassandraTestContext().annotation;
+        assertThat(writeReplicas).hasSize(expectedRangeMapping.get("datacenter1").size());
```

Comment:
Having trouble wrapping my head around why writeReplicas would always be the size of only the first dc here - can you clarify (maybe add a comment) as to why this is the case?

Reply:
This validates the number of ranges match the expected no. ranges.